external/magmahelper/make.dpbuild

```diff
@@ -20,27 +20,23 @@ vpath % $(SRCDIR)
 
 CSRC = $(notdir $(wildcard $(SRCDIR)/*.c))
 F90SRC = $(notdir $(wildcard $(SRCDIR)/*.f90))
-MAGMAOBJS = $(patsubst %.c,%.o,$(CSRC)) $(patsubst %.f90,%.o,$(F90SRC))
+CDEVICEOBJS = $(patsubst %.c,%.o,$(CSRC))
+FDEVICEOBJS = $(patsubst %.f90,%.o,$(F90SRC))
 
+
+.PHONY: all
 all: libCountDevice.a
-	touch -r $< $(STAMPFILE)
+	touch -r $<  $(OBJDIR)/libmagmahelper.a $(STAMPFILE)
 
-libCountDevice.a: $(MAGMAOBJS)
+libCountDevice.a: $(FDEVICEOBJS) $(CDEVICEOBJS)
 	ar rcs $@ $^
 
 PHONY: clean
 clean:
 	rm -f *.o *.mod *.a
 
 %.o: %.f90
-	$(FXX) $(FXXOPT) -o $@ -c $<
+	$(FXX) -c $<
 
 %.o: %.c
-	$(CC) $(CCOPT) -I$(MAGMADIR)/include -I$(CUDADIR)/include -o $@ -c $<
-
-_mod_%:
-	@[ -f $< ] && touch $@
-
-### Local Variables:
-### mode:makefile
-### End:
+	$(CC) $(CCOPT) $(LDFLAGS) -c $<
```

Comment:
Is there a reason for removing the include paths? If CUDA and magma are not installed in the standard include path (probably the case for most machine managers location of choice) this can't be set to find them.

Reply:
if you include make.config, this includes make.arch and I include magma and CUDA include paths in make.arch. However, if you prefer to do it this way you can leave it like that. Perhaps you are right and it's safer to include them here too.